src/juju/types.ts

```diff
@@ -32,11 +32,33 @@ export interface UnitData {
 export type AllWatcherDelta =
   | ["action", "change", ActionChangeDelta]
   | ["application", "change", ApplicationChangeDelta]
+  | ["application", "remove", ApplicationChangeDelta]
   | ["charm", "change", CharmChangeDelta]
+  | ["charm", "remove", CharmChangeDelta]
   | ["unit", "change", UnitChangeDelta]
+  | ["unit", "remove", UnitChangeDelta]
   | ["machine", "change", MachineChangeDelta]
+  | ["machine", "remove", MachineChangeDelta]
   | ["model", "change", ModelChangeDelta]
-  | ["relation", "change", RelationChangeDelta];
+  | ["relation", "change", RelationChangeDelta]
+  | ["relation", "remove", RelationChangeDelta];
+
+export type DeltaEntityTypes =
```

Comment:
These could be more useful as an enum that you could then use when defining the `AllWatcherDelta` tuples above.